lib/AST/Expr.cpp

```diff
@@ -1771,7 +1771,7 @@ Expr *CallExpr::getDirectCallee() const {
 }
 
 void ExplicitCastExpr::setCastType(Type type) {
-  CastTy->setType(MetatypeType::get(type));
```

Comment:
I believe `MetatypeType` is used here to maintain the invariant that a `TypeExpr` is always typed as a `MetatypeType`.

Reply:
Yes, please don't do this without a more holistic modeling change. Meta-metatypes *are* the types of cast expressions with metatypes as destinations.

Reply:
I'm switch the approach to avoid the TypeExpr type entirely.